src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.cs

```diff
@@ -1770,28 +1773,27 @@ private bool SkipMultiLineComment(ReadOnlySpan<byte> localBuffer, out int idx)
                 }
                 if (foundIdx != 0 && localBuffer[foundIdx + idx - 1] == JsonConstants.Asterisk)
                 {
-                    idx += foundIdx;
+                    idx += foundIdx - 1;
                     break;
                 }
                 idx += foundIdx + 1;
             }
 
-            Debug.Assert(idx >= 1);
-
             // Consume the /* and */ characters that are part of the multi-line comment.
             // Since idx is pointing at right after the final '*' (i.e. before the last '/'), we don't need to count that character.
             // Hence, we increment consumed by 3 (instead of 4).
-            _consumed += 4 + idx - 1;
+            _consumed += 4 + idx;
 
-            (int newLines, int newLineIndex) = JsonReaderHelper.CountNewLines(localBuffer.Slice(0, idx - 1));
+            (int newLines, int newLineIndex) = JsonReaderHelper.CountNewLines(localBuffer.Slice(0, idx));
             _lineNumber += newLines;
             if (newLineIndex != -1)
             {
-                _bytePositionInLine = idx - newLineIndex;
+                // Exclude linefeed while resetting position
+                _bytePositionInLine = 2 + idx - (newLineIndex + 1);
```

Comment:
What does the `+ 1` mean here?

Reply:
It means excluding the line feed. Will add more text.

Reply:
My bad. I will revert this. The error in which this value is used says 'byte position' and I confused it with 'char position' and always found it to be one less.

Reply:
> My bad. I will revert this. The error in which this value is used says 'byte position' and I confused it with 'char position' and always found it to be one less.

Please ignore the above. The new logic is correct and neither was there any bug in previous logic. Reason being that the meaning of `idx` has changed. Previously it pointed to the `/` in the end-of-comment `*/` since comment text used to include delimiters as well. The delimiters are now excluded. As per new logic `idx` points to just after the last character of comment i.e. the `*` in `*/`. Hence the need to add `1`.

```C#
// Old logic where idx pointed to terminal / in comment
_bytePositionInLine = idx - newLineIndex;

// New logic where idx points to * preceding the terminal / in comment.
// Rearranging, the new expression can be written as,
_bytePositionInLine = idx + 1 - newLineIndex;
// I will rewrite above code as below to avoid overflows if any
_bytePositionInLine = idx - newLineIndex + 1;
```
Example:
```
Comment text where N = \n:
/*abcNdef*/
012345678901
      012345
Previously:
idx = 10, newline = 5, byte posn = idx - newline = 5
New logic:
idx =  9, newline = 5, byte posn = idx - newline + 1 = 5
```